pallets/frequency-tx-payment/src/tests.rs

```diff
@@ -664,3 +698,70 @@ fn compute_capacity_fee_successful() {
 			assert_eq!(fee, 26);
 		});
 }
+
+#[test]
+fn pay_with_capacity_batch_all_errors_when_transaction_amount_exceeds_maximum() {
+	let balance_factor = 10;
+	let too_many_calls = vec![
+		RuntimeCall::Balances(BalancesCall::transfer { dest: 2, value: 100 }),
+		RuntimeCall::Balances(BalancesCall::transfer { dest: 2, value: 100 }),
+		RuntimeCall::Balances(BalancesCall::transfer { dest: 2, value: 100 }),
+	];
+
+	ExtBuilder::default()
+		.balance_factor(balance_factor)
+		.base_weight(Weight::from_ref_time(5))
+		.build()
+		.execute_with(|| {
+			let who = 1u64;
+
+			assert_noop!(
+				FrequencyTxPayment::pay_with_capacity_batch_all(
+					RuntimeOrigin::signed(who),
+					too_many_calls
+				),
+				Error::<Test>::BatchedCallAmountExceedMaximum
+			);
+		});
+}
+
+#[test]
+fn pay_with_capacity_batch_all_transactions_will_all_fail_if_one_fails() {
+	let balance_factor = 10;
+
+	ExtBuilder::default()
+		.balance_factor(balance_factor)
+		.base_weight(Weight::from_ref_time(5))
+		.build()
+		.execute_with(|| {
+			let origin = 1u64;
+			let successful_balance_transfer_call =
+				RuntimeCall::Balances(BalancesCall::transfer { dest: 2, value: 100 });
+
+			let balance_transfer_call_insufficient_funds =
+				RuntimeCall::Balances(BalancesCall::transfer { dest: 2, value: 100000000 });
+
+			let token_balance_before_call = Balances::free_balance(origin);
+
+			let calls_to_batch =
+				vec![successful_balance_transfer_call, balance_transfer_call_insufficient_funds];
+
+			let result = FrequencyTxPayment::pay_with_capacity_batch_all(
+				RuntimeOrigin::signed(origin),
+				calls_to_batch,
+			);
+
+			assert!(match result {
+				Err(DispatchErrorWithPostInfo { .. }) => {
+					true
+				},
+				_ => {
+					false
+				},
+			});
+
+			let token_balance_after_call = Balances::free_balance(origin);
+
+			assert_eq!(token_balance_before_call, token_balance_after_call);
+		});
+}
```

Comment:
Can you please add a couple of test for the extrinsic pay_with_capacity_batch_all? One Happy path and the other for the error `BatchedCallAmountExceedMaximum`?

Reply:
Transaction amount maximum  is above this one,
Added the happy path test